### PR TITLE
Information and download button for each chart

### PR DESCRIPTION
**Pull request by tinashechiraya**
[Screencast from 19-10-2023 00:33:45.webm](https://github.com/kartoza/sawps/assets/70011086/6dcdc58a-6c61-42a9-9054-c54e744ed177)

Description
each chart now has an information and download icon
the information icon when hovered over will show a tooltip which will display the information on the chart ,
the information is still to be provided by @LunaAsefaw 
the download icon downloads the individual chart
I have placed both icons at the bottom right as requested

I have added the topper to the charts page 
Screenshot from 2023-10-19 13-40-29

and this a sample chart downloaded with the download  icons
[Cheetah-3.pdf](https://github.com/kartoza/sawps/files/13043700/Cheetah-3.pdf)

### CODE REVIEW

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 7 min 23 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 9 min 58 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 7 min 16 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 7 min 36 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 8 min 36 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 2 min 8 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 7 min 50 sec

File: django_project/frontend/src/containers/MainPage/Metrics/index.tsx [hunk 1]
```diff
@@ -94,10 +130,27 @@ const Metrics = () => {
...
+    // Remove '/charts' from the current URL
+    currentLocation = currentLocation.replace('/charts', '');
+
+    // Replace 'localhost:8000' with the current full URL
+    const imageUrlWithDomain = imageUrl.replace('http://localhost:8000', currentLocation);
+
+    const informationIconUrl = 'http://localhost:8000/static/images/icons/information.svg';
+
+    // Replace 'localhost:8000' with the current full URL
+    const informationIconUrlWithDomain = informationIconUrl.replace('http://localhost:8000', currentLocation);
```
Comment (dimasciput) [inline django_project/frontend/src/containers/MainPage/Metrics/index.tsx:151]: @tinashechiraya can you just use `/static/...` url? Without the domain.

File: django_project/frontend/src/containers/MainPage/Metrics/index.tsx [hunk 2]
```diff
@@ -193,16 +245,136 @@ const Metrics = () => {
...
         pdf.save('metrics.pdf');
+        setIconsHidden(false);
     }
 
+
+    type Constants = {
+        [key: string]: boolean;
+    };
+    
+    const constants: Constants = {
```
Comment (dimasciput) [inline django_project/frontend/src/containers/MainPage/Metrics/index.tsx:256]: Isn't this from a different PR?

**Review by dimasciput — CHANGES_REQUESTED** [review]

File: django_project/frontend/src/containers/MainPage/Metrics/index.tsx (hunk 1 above)
Comment (tinashechiraya) [reply django_project/frontend/src/containers/MainPage/Metrics/index.tsx]: @dimasciput I have resolved this

File: django_project/frontend/src/containers/MainPage/Metrics/index.tsx (hunk 2 above)
Comment (tinashechiraya) [reply django_project/frontend/src/containers/MainPage/Metrics/index.tsx:251]: @dimasciput it's from the previous PR . I think I pushed them at the same time

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 8 min 8 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 8 min 37 sec

**Comment by dimasciput** [comment]
Please update the PR

**Comment by tinashechiraya** [comment]
> Please update the PR

@dimasciput I have done so

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 7 min 20 sec

**Comment by tinashechiraya** [comment]
@dimasciput can you merge this?

**Comment by dimasciput** [comment]
@tinashechiraya I will update this PR first since it has a lot of conflicts.

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** FAILURE
**Build duration:** 56 sec

**Pull request closed without merging**